text/0193-removal-of-type-unions.md

```diff
@@ -0,0 +1,484 @@
+---
+feature name: removal-of-type-unions
+start date: 2020-07-06
+rfc pr: 194
+related issue: 193
+---
+
+# Summary
+
+_Type unions_ is a feature of the TypeScript language that allows typing values as "one of several candidate types".
+This is achieved by specifying a `|`-delimited list of candidate types (e.g: `Foo | Bar`). While this feature is common
+in dynamic languages (**TypeScript**, **Python**, ...), it is almost always absent from statically typed languages
+(**Java**, **C#**, ...). In those languages, the impossibility to represent _type unions_ natively results in degraded
+developer experience.
+
+This RFC proposes to remove all usage of _type unions_ from the AWS CDK, and eventually the removal of the feature from
+the `jsii` type model.
+
+# README
+
+Use of TypeScript _type unions_ is prohibited by `jsii` due to the lack of support of this language feature in several
+of the target languages supported (or planned to be supported) by `jsii`: **C#**, **Go**, **Java**, and probably more
+languages to come. Proper support of _type unions_ in those languages requires generating code that produces a degraded
+developer experience in those languages.
+
+The recommended substitution pattern for APIs where _type unions_ would be the natural candidate in dynamic languages is
```

Comment:
I kind of feel that a proposal to remove a feature should include enough evidence of the feasibility of an implementation that does not use said feature?

Reply:
I see. Perhaps make this section a bit more concise (e.g reference docs or a GitHub about implementing union-like classes) because it buries the important stuff.